src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/TelemetryClientFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector;
+using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse;
+using Microsoft.ApplicationInsights.WindowsServer;
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// Creates a <see cref="TelemetryClient"/> for use by the <see cref="ApplicationInsightsLogger"/>. 
+    /// </summary>
+    [CLSCompliant(false)]
+    public class TelemetryClientFactory : ITelemetryClientFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="TelemetryClient"/>. 
+        /// </summary>
+        /// <returns>The <see cref="TelemetryClient"/> instance.</returns>
+        public TelemetryClient Create(string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings)
+        {
+            TelemetryConfiguration config = InitializeConfiguration(instrumentationKey, samplingSettings);
+
+            TelemetryClient client = new TelemetryClient(config);
+
+            string assemblyVersion = GetAssemblyFileVersion(typeof(JobHost).Assembly);
+            client.Context.GetInternalContext().SdkVersion = $"webjobs: {assemblyVersion}";
+
+            return client;
+        }
+
+        internal static string GetAssemblyFileVersion(Assembly assembly)
+        {
+            AssemblyFileVersionAttribute fileVersionAttr = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>();
+            return fileVersionAttr?.Version ?? LoggingConstants.Unknown;
+        }
+
+        internal static TelemetryConfiguration InitializeConfiguration(string instrumentationKey,
```

Comment:
How do we plan to expose configuration options for http logging, perf counters, adaptive sampling in functions?

Also, I think we need to be careful if we enable adaptive sampling transparently by default (and perhaps other options) - it might surprise customers.